Client/Frontend/Browser/Search/BraveSearchPromotionCell.swift

```diff
@@ -0,0 +1,175 @@
+    $0.backgroundColor = .braveOrange
+    $0.snp.makeConstraints { make in
+      make.width.greaterThanOrEqualTo(120)
+    }
+    $0.contentEdgeInsets = UIEdgeInsets(top: 8, left: 12, bottom: 8, right: 12)
+    $0.setContentHuggingPriority(.defaultHigh, for: .vertical)
+    $0.setContentCompressionResistancePriority(.required, for: .vertical)
+  }
+  
+  private let dismissButton = UIButton().then {
+    $0.titleLabel?.font = UIFont.systemFont(ofSize: 14, weight: .semibold)
+    $0.titleLabel?.minimumScaleFactor = 0.75
+    $0.setTitleColor(.braveOrange, for: .normal)
+    $0.setTitle(
+      Preferences.BraveSearch.braveSearchPromotionCompletionState.value != BraveSearchPromotionState.maybeLaterUpcomingSession.rawValue ?
+        Strings.BraveSearchPromotion.braveSearchPromotionBannerMaybeLaterButtonTitle :
+        Strings.BraveSearchPromotion.braveSearchPromotionBannerDismissButtonTitle,
+      for: .normal)
+    $0.backgroundColor = .clear
+    $0.setContentHuggingPriority(.defaultHigh, for: .vertical)
+    $0.setContentCompressionResistancePriority(.required, for: .vertical)
+  }
+
+  override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+    super.init(style: style, reuseIdentifier: reuseIdentifier)
+    
+    backgroundColor = .secondaryBraveBackground
+    selectionStyle = .none
+
+    contentView.addSubview(promotionContentView)
+    promotionContentView.snp.makeConstraints {
+      $0.leading.equalTo(safeArea.leading).inset(8)
+      $0.trailing.equalTo(safeArea.trailing).inset(8)
+      $0.top.equalTo(safeArea.top)
+      $0.bottom.equalTo(safeArea.bottom)
+    }
+    
+    [tryButton, dismissButton].forEach(buttonsStackView.addArrangedSubview(_:))
+    
+    mainStackView.addStackViewItems(
+      .view(titleLabel),
+      .customSpace(8.0),
+      .view(bodyLabel),
+      .customSpace(16.0),
+      .view(buttonsStackView)
+    )
+    
+    buttonsStackView.snp.makeConstraints {
+      $0.height.equalTo(tryButton.snp.height)
+    }
+    
+    promotionContentView.addSubview(mainStackView)
+    promotionContentView.addSubview(promotionalImageView)
+
+    mainStackView.snp.makeConstraints {
+      $0.leading.top.equalToSuperview().offset(16)
+      $0.bottom.equalToSuperview().offset(-16)
+      $0.trailing.equalTo(promotionalImageView.snp.leading)
+    }
+    
+    promotionContentView.snp.makeConstraints {
+      $0.edges.equalToSuperview()
+    }
+    
+    promotionalImageView.snp.makeConstraints {
+      $0.top.bottom.equalToSuperview()
+      $0.trailing.equalToSuperview()
+    }
+
+    tryButton.addTarget(self, action: #selector(tryAction), for: .touchUpInside)
+    dismissButton.addTarget(self, action: #selector(dismissAction), for: .touchUpInside)
+  }
+
+  @available(*, unavailable)
+  required init(coder: NSCoder) { fatalError() }
+
+  @objc func tryAction() {
+    trySearchEngineTapped?()
+  }
+
+  @objc func dismissAction() {
+    dismissTapped?()
+  }
+}
+
+extension UIColor {
```

Comment:
Should this be a `private` extension? `borderColor` / `backgroundColor` are a bit generic unless they will be used elsewhere

Reply:
Fix at https://github.com/brave/brave-ios/pull/5432/commits/e13ac5a7e81aa9ce8df9852cad0f9b34774ca6f9